Don't try to open syslog socket until it is really needed.  When used in
docker containers, this prevents pies from spitting spurious diagnostic
message about not being able to open /dev/log.  Hopefully this will be
fixed upstream some day.

diff --git a/src/diag.c b/src/diag.c
index 112404f..ddbad64 100644
--- a/src/diag.c
+++ b/src/diag.c
@@ -144,8 +144,8 @@ diag_setup (int flags)
 {
   if (flags)
     diag_output = flags;
-  if (diag_output & DIAG_TO_SYSLOG)
-    pies_syslog_open ();
+  if (!(diag_output & DIAG_TO_SYSLOG))
+    pies_syslog_close ();
 }
 
 void
